File: ruby/02-switch.rb
require 'rpi_gpio'

# Run `gpio readall` to confirm BCM pin number(s)
pin_in = 18 # button

RPi::GPIO.set_numbering(:bcm)
RPi::GPIO.set_warnings(0)
RPi::GPIO.setup(pin_in, as: :input, pull: :down)

begin
  loop do
    if RPi::GPIO.high?(pin_in) # button input state is "on"
      puts 'Button pressed'
      sleep(0.2)
    end
  end
rescue Interrupt
  puts 'Done'
ensure
  RPi::GPIO.clean_up
end

File: node/02-switch.js
var Gpio = require('onoff').Gpio,
    button = new Gpio(18, 'in', 'both');  // Run `gpio readall` to confirm BCM pin number(s)

// Watch for button state to change from "0" to "1"
button.watch(function(err, value) {
    if (err) exit();
    if (value == 1) {
        console.log('Button pressed');
    }
});

function exit() {
    // Cleanup
    button.unexport();
    process.exit();
}

// Exit on kill signal (Ctrl+C)
process.on('SIGINT', exit);